src/oc_erchef/apps/oc_chef_wm/src/chef_wm_depsolver.erl

```diff
@@ -278,22 +281,64 @@ wm_halt(Code, Req, State, ErrorData, LogMsg) ->
 %% removing large fields such as long_description and attributes in
 %% the metadata that are not required by chef-client
 assemble_response(Req, #base_state{server_api_version = ApiVersion} = State, CookbookVersions) ->
+
     case oc_chef_wm_base:check_cookbook_authz(CookbookVersions, Req, State) of
         ok ->
-            %% We iterate over the list again since we only want to construct the s3urls
-            %% if the authz check has succeeded.  We use a minimal version of the
-            %% cookbook which has just enough information for chef-client to run
-            JsonList = {
-                    [ { CBV#chef_cookbook_version.name,
-                       chef_cookbook_version:minimal_cookbook_ejson(CBV, chef_wm_util:base_uri(Req), ApiVersion) }
-                      || CBV <- CookbookVersions ]
-                    },
-            CBMapJson = chef_json:encode(JsonList),
-            {true, wrq:append_to_response_body(CBMapJson, Req), State};
+            case make_json_list(CookbookVersions, chef_wm_util:base_uri(Req), ApiVersion) of
+                {error, busy} ->
+                  % Force backoff until the cache catches up with demand. Occurs when caching is enabled
+                  % and the cache message queue is overloaded.
```

Comment:
Also occurs when we give up on waiting for another process that has claimed a given cache key to complete its work.

Reply:
Do we want to add this^^ to the comment?

Reply:
```
                  % Force backoff until the cache catches up with demand. Occurs when caching is enabled
                  % and the cache message queue is overloaded. Also occurs when we give up on waiting for
                  % another process that has claimed a given cache key to complete its work.
```